lib/EnzymeCore/src/rules.jl

```diff
@@ -18,6 +18,17 @@ the annotated function arguments.
 """
 function forward end
 
+"""
+    Config{NeedsPrimal, NeedsShadow, Width, Overwritten}
+    ConfigWidth{Width} = Config{<:Any,<:Any, Width}
+
+Configuration type to dispatch on in custom reverse rules (see [`augmented_primal`](@ref) and [`reverse`](@ref)).
+* `NeedsPrimal` and `NeedsShadow` are boolean values specifying whether the primal and shadow (resp.) should be returned. 
+* `Width` is an integer that specifies the number of adjoints/shadows simultaneously being propagated.
+* `Overwritten` is ??? (TODO) 
+
+Getters for the four type parameters are provided by `needs_primal`, `needs_shadow`, `width`, and `overwritten`.
+"""
 struct Config{NeedsPrimal, NeedsShadow, Width, Overwritten} end
```

Comment:
Wonder if this is better named `ReverseConfig`

Reply:
That change is fine with me.

Reply:
Shall I do that in a PR immediately after this gets merged? To keep this change purely docs